### DISCUSSION TITLE
Keyboard Input doesn't work properly under linux Wayland

### DISCUSSION THREAD

**Discussion by itm154** [ppy/osu#23203] [Q&A] [answered]
Things like the beatmap search and settings search does take in keyboard input but only the alpha keys, but when in-game, clicks are not working, and also modifier such as ESC, CTRL-O, Enter, Backspace does not work (even in search fields). Going into the key configuration page also does not take in inputs (changing keys). $SDL_VIDEODRIVER has also been set to wayland.

I did not have this issue before (same version of the game) but I have absolutely no idea what caused it. I have an IME (fcitx-5) but it works perfectly fine before.

Tablet inputs works fine with OTD (non in-game one)

Running under Hyprland on Arch Linux  (6.2.10-arch1-1) and the game is installed from the [AUR](https://aur.archlinux.org/packages/osu-lazer-bin)

[database.log](https://github.com/ppy/osu/files/11233627/database.log)
[input.log](https://github.com/ppy/osu/files/11233628/input.log)
[network.log](https://github.com/ppy/osu/files/11233629/network.log)
[performance.log](https://github.com/ppy/osu/files/11233630/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11233631/runtime.log)

**Comment by itm154** [comment, marked as answer]
Issue is solved by reinstalling the game. Probably caused by corrupted files

Little note when reinstalling, copy the 'files' folder and client.realm file so Lazer can detect all of your previous files even from a clean install. Took me a while to figure out

**Reply by Samir2zk** [reply]
Thanks bro it worked! Running same setup as you.

**Reply by skaldebane** [reply]
Thanks, works on Fedora 38 with the same issue!

**Reply by chayleaf** [reply]
To anyone else who stumbles on this, you can simply change `input.json` - for some reason, I had this `{"$type":"osu.Framework.Input.Handlers.Keyboard.KeyboardHandler,osu.Framework","Description":"Keyboard","IsActive":true,"Enabled":false}` in it (notice the `"Enabled":false`)

**Reply by peppy** [reply]
For what it's worth this should be fixed in the latest release.
